es6-persistence/src/main/java/com/netflix/conductor/dao/es6/index/ElasticSearchDAOV6.java

```diff
@@ -467,20 +468,31 @@ private void updateWithRetry(UpdateRequest request, String operationDescription)
         return search(query, start, count, sort, freeText, TASK_DOC_TYPE);
     }
 
-    @Override
-    public void removeWorkflow(String workflowId) {
+    private void removeDocument(String indexName, String docType, String docId) {
         try {
-            DeleteRequest request = new DeleteRequest(workflowIndexName, WORKFLOW_DOC_TYPE, workflowId);
+            DeleteRequest request = new DeleteRequest(indexName, docType, docId);
             DeleteResponse response = elasticSearchClient.delete(request).actionGet();
-            if (response.getResult() == DocWriteResponse.Result.DELETED) {
-                logger.error("Index removal failed - document not found by id: {}", workflowId);
+
+            if (response.getResult() == DocWriteResponse.Result.NOT_FOUND) {
+                logger.error(docType + " removal failed - document not found by id: {}", docId);
             }
         } catch (Throwable e) {
-            logger.error("Failed to remove workflow {} from index", workflowId, e);
+            logger.error("Failed to remove " + docType + " {} from index", docId, e);
             Monitors.error(className, "remove");
         }
     }
 
+    @Override
+    public void removeWorkflow(String workflowId) {
+        removeDocument(workflowIndexName, WORKFLOW_DOC_TYPE, workflowId);
+
+        List<String> taskIds = searchTasks("", "workflowId:\"" + workflowId + "\"", 0, 100, Collections.emptyList()).getResults();
```

Comment:
So, the searchTasks parameters are start=0, and count=100. What if the workflow has more than 100 tasks?